File: articles/data-measurement-reproducibility-life-sciences/python/measurement_quality_summary.py
"""
Measurement quality summary workflow.

Run:
    python python/measurement_quality_summary.py
"""

from pathlib import Path

import pandas as pd

from reproducibility_core import measurement_quality_summary


ARTICLE_DIR = Path(__file__).resolve().parents[1]
DATA_PATH = ARTICLE_DIR / "data" / "measurements.csv"


def main() -> None:
    data = pd.read_csv(DATA_PATH)
    summary = measurement_quality_summary(data)

    print(pd.DataFrame([summary.__dict__]).round(5).to_string(index=False))


if __name__ == "__main__":
    main()

File: articles/data-measurement-reproducibility-life-sciences/r/measurement_quality_summary.R
# Measurement quality summary workflow.

data_path <- file.path("data", "measurements.csv")
if (!file.exists(data_path)) {
  data_path <- file.path("..", "data", "measurements.csv")
}

measurements <- read.csv(data_path)

values <- suppressWarnings(as.numeric(measurements$measurement_value))
pass_values <- values[measurements$qc_flag == "pass" & !is.na(values)]

summary_df <- data.frame(
  n_total = nrow(measurements),
  n_missing = sum(is.na(values)),
  n_pass = sum(measurements$qc_flag == "pass"),
  n_review = sum(measurements$qc_flag == "review"),
  n_fail = sum(measurements$qc_flag == "fail"),
  completeness_rate = 1 - sum(is.na(values)) / nrow(measurements),
  qc_pass_rate = sum(measurements$qc_flag == "pass") / nrow(measurements),
  mean_value = mean(pass_values),
  sd_value = sd(pass_values),
  coefficient_of_variation = sd(pass_values) / mean(pass_values)
)

print(round(summary_df, 5))
